File: solutions/java/high-scores/1/src/main/java/HighScores.java
import java.util.List;
import java.util.ArrayList;
import java.util.Collections;

class HighScores {

    private List<Integer> s;

    public HighScores(List<Integer> highScores) {
        s = highScores;
    }

    List<Integer> scores() {
        return this.s;
    }

    Integer latest() {
        return s.get(s.size()-1);
    }

    Integer personalBest() {
        int best = this.s.get(0);
        for(int i=1;i<this.s.size();i++){
            if (this.s.get(i)>best){
                best = this.s.get(i);
            }
        }
        return best;
    }

    List<Integer> personalTopThree() {
        if (this.s.size()<=3){
            Collections.sort(this.s,Collections.reverseOrder());
            return this.s;
        }
        int best1 = -100;
        int best2 = -100;
        int best3 = -100;
        for(int i=0;i<this.s.size();i++){
            if (this.s.get(i)>best1){
                best3 = best2;
                best2 = best1;
                best1 = this.s.get(i);
            }else if(this.s.get(i)>best2){
                best3 = best2;
                best2 = this.s.get(i);
            }else if(this.s.get(i)>best3){
                best3 = this.s.get(i);
            }
        }
        List<Integer> a = new ArrayList<>();
        a.add(best1);
        a.add(best2);
        a.add(best3);
        return a;
    }

}

File: solutions/c/high-scores/1/high_scores.c
#include "high_scores.h"

int32_t latest(const int32_t *scores, size_t scores_len){
    return scores[scores_len-1];
}

int32_t personal_best(const int32_t *scores, size_t scores_len){
    size_t i = 1;
    int32_t best = scores[0];
    while (i<scores_len){
        if (scores[i]>best){
            best = scores[i];
        }
        i++;
    }
    return best;
}

size_t personal_top_three(const int32_t *scores, size_t scores_len, int32_t *output){
    int32_t best = INT32_MIN;
    int32_t best2 = INT32_MIN;
    int32_t best3 = INT32_MIN;
    size_t i = 0;
    while (i<scores_len){
        if (scores[i]>best){
            best3 = best2;
            best2 = best;
            best = scores[i];
        }else if (scores[i]>best2){
            best3 = best2;
            best2 = scores[i];
        }else if (scores[i]>best3){
            best3 = scores[i];
        }
        i++;
    }
    size_t cnt = 0;
    if (scores_len>0){
        output[0] = best;
        cnt++;
    }
    if (scores_len>1){
        output[1] = best2;
        cnt++;
    }
    if (scores_len>2){
        output[2] = best3;
        cnt++;
    }
    return cnt;
}